src/workerd/io/actor-sqlite.c++

```diff
@@ -449,27 +455,34 @@ kj::Promise<void> ActorSqlite::commitImpl(ActorSqlite::PrecommitAlarmState preco
   // Notify any merged commitImpl() requests that the db persistence completed.
   fulfiller->fulfill();
 
-  // If the db state is now later than the in-flight scheduled alarms, issue a request to update
-  // it to match the db state.
-  if (willFireEarlier(alarmScheduledNoLaterThan, alarmStateForCommit)) {
-    if (debugAlarmSync) {
-      KJ_LOG(WARNING, "NOSENTRY DEBUG_ALARM: Moving alarm later", "sqlite_has",
-          logDate(alarmStateForCommit), "alarmScheduledNoLaterThan",
-          logDate(alarmScheduledNoLaterThan));
+  // If another commit modified the alarm while we were async, skip post-commit alarm sync.
+  //
+  // We do this for a few reasons:
+  //  1. The other commit will handle its own alarm sync
+  //  2. Post-commit syncs are inherently optional (the alarm will self-correct)
+  //  3. This coalesces redundant alarm updates for better performance
```

Comment:
Should add:

> 4. This avoids race conditions where a later commit moved the alarm earlier, requiring a pre-commit alarm update, and this update may have already been made before we get here.

Reply:
https://github.com/cloudflare/workerd/compare/362c2fc34fe0c3c09ed88b841e0d7e9e95696695..73569aa44a9af2ae5a3afcbea2d51962f8a2a02a#diff-e0f2e99f58d22f6880187a9de379c249811338370351ce2bea41b905865193d9R464 done.